ethereum/p2p/src/test/java/tech/pegasys/pantheon/ethereum/p2p/discovery/internal/MockTimerUtil.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.p2p.discovery.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MockTimerUtil implements TimerUtil {
+  private final AtomicLong nextId = new AtomicLong(0);
+  private final Map<Long, TimerHandler> timerHandlers = new HashMap<>();
+  private final Map<Long, TimerHandler> periodicHandlers = new HashMap<>();
+
+  @Override
+  public long setPeriodic(final long delay, final TimerHandler handler) {
+    long id = nextId.incrementAndGet();
+    periodicHandlers.put(id, handler);
```

Comment:
Do we have any test cases that depend on different delays between peers for timers or handlers?

Reply:
We're not doing anything very complicated with the timers.  We can always extend this in the future if we need to.